### DISCUSSION TITLE
Flutter框架介紹

### DISCUSSION THREAD

**Discussion by quan0715** [1101-GS4719-C/Python-Community-GS4719-C#1325] [科技新知]
## Flutter框架介紹
Flutter是一款由Google開發的跨平台UI框架，使用Dart語言編寫，可以在iOS、Android、Web、Windows、macOS、Linux等多個平台上運行。Flutter擁有優秀的性能表現、豐富的組件庫、簡單易用的API等優點，越來越受到開發者的關注和喜愛。

### 特點
快速開發
Flutter具有豐富的組件庫和強大的開發工具，讓開發者可以快速地構建出高品質的UI和交互體驗，縮短開發週期，提高開發效率。

### cross-platform 跨平台開發
Flutter支持多個平台，一套代碼即可運行在iOS、Android、Web、Windows、macOS、Linux等不同的設備上，大大減少了開發成本和維護成本。

### Hot-reload
Flutter擁有獨特的熱重載功能，修改代碼後不需要重新啟動應用程序，即可實時看到變化，大大提高了開發效率。

### 高性能
Flutter使用自有的渲染引擎Skia來實現高性能的UI渲染，並使用Dart的Just-in-Time（JIT）編譯技術實現快速的開發迭代和調試，同時也支持Ahead-of-Time（AOT）編譯技術，提高了應用程序的運行速度和性能表現。

### UI KIT & package
Flutter擁有豐富的組件庫，包括Material Design和Cupertino風格的UI組件，以及動畫、手勢、佈局等多種功能組件，開發者可以輕鬆地構建出美觀、流暢的應用程序。

### 總結
Flutter是一款強大的跨平台UI框架，具有快速開發、跨平台、熱重載、高性能等優點，擁有豐富的組件庫和活躍的社區支持，是現代移動應用程序開發的理想選擇。
更多資訊請去 [官網](https://flutter.dev/) 觀看更多資訊

### 推薦心得
跨平台開發實在是太香啦，寫一份code通吃全部的平台不香嗎？又是現在google爸爸力捧的潛力框架，不管是做design、後端服務等等都有強大的google 生態系支援，透過此框架可以加速你的產品開發週期，快速展示你的成果，目前已經擠身主流開發前端框架，唯獨學習成本偏高，但鼓勵多看官方文檔案與官方教學影片，會大大加速你的學習速度。

**Comment by ziihui** [comment]
沒資訊量ㄎㄎ
